lua/wire/sh_wirenet.lua

```diff
@@ -0,0 +1,194 @@
+--[[
+	Consolidated networking library for Wiremod.
+	Trivial Networking - A wrapper for sending very small net messages that don't need a whole networkstring.
+]]
+
+local Net = WireLib.Net or {}
+WireLib.Net = Net
+
+-- Trivial Networking
+local SIZE = 10 -- 1023 names
+
+local registered_handlers = {}
+
+local update_handlers
+
+local function msg_handler(len, ply)
+	local handler = registered_handlers[net.ReadUInt(SIZE)]
+	if handler then
+		handler(len, ply)
+	else
+		error("WireLib.Net tried to receive message that is not registered.")
+	end
+end
+
+net.Receive("wirelib_net_message", msg_handler)
+
+if SERVER then
+	util.AddNetworkString("wirelib_net_message")
+
+	local handler_queue = {}
+	local queue_handler_firstidx, queue_handler_lastidx
+
+	local function queue_handler_flush(ply)
+		local restart, lastidx = false, queue_handler_lastidx
+		if queue_handler_firstidx then
+			net.Start("wirelib_net_message")
+
+				local data, len = "", 0
+				for i, v in ipairs(handler_queue) do
+					len = len + #v + 1
+					data = data .. v .. "\0"
+					if len > 2048 then
+						lastidx = i
+						restart = true
+						break
+					end
+				end
+
+				net.WriteUInt(0, SIZE)
+				net.WriteUInt(queue_handler_firstidx, SIZE)
+				net.WriteUInt(lastidx, SIZE)
+
+				data = util.Compress(data)
+				local datalen = #data
+				net.WriteUInt(datalen, 16)
+				net.WriteData(data, datalen)
+
+			if ply then net.Send(ply) else net.Broadcast() end
+		end
+
+		if restart and queue_handler_firstidx < queue_handler_lastidx then
+			queue_handler_firstidx = lastidx + 1
+			timer.Create("wirelib_net_flush", 0, 2, queue_handler_flush)
+		else
+			handler_queue = {}
+			queue_handler_firstidx, queue_handler_lastidx = nil, nil
+			timer.Remove("wirelib_net_flush")
+		end
+	end
+	local function queue_handler_update(idx, name)
+		if queue_handler_firstidx then
+			if queue_handler_lastidx + 1 == idx then
+				handler_queue[#handler_queue + 1] = name
+				queue_handler_lastidx = idx
+			else
+				queue_handler_flush()
+
+				queue_handler_firstidx, queue_handler_lastidx = idx, idx
+				handler_queue[1] = name
+
+				timer.Adjust("wirelib_net_flush", 1, 2)
+			end
+
+		else
+			queue_handler_firstidx, queue_handler_lastidx = idx, idx
+			handler_queue[1] = name
+
+			timer.Create("wirelib_net_flush", 1, 2, queue_handler_flush)
```

Comment:
Can we import netstream yet? This needs to stop

Reply:
Would I just copy netstream from Starfall into Wiremod?

Reply:
I've been asking for netstream to be merged for ages, go ahead